lib/dust.js

```diff
@@ -565,9 +585,36 @@
     return this;
   };
 
+  /**
+   * Pipes to a WritableStream. Note that backpressure isn't implemented,
+   * so we just write as fast as we can.
+   * @param stream {WritableStream}
+   * @return self
+   */
   Stream.prototype.pipe = function(stream) {
+    if(typeof stream.write !== 'function' ||
+       typeof stream.end !== 'function') {
+      dust.log('Incompatible stream passed to `pipe`', WARN);
+      return this;
+    }
+
+    var destEnded = false;
+
+    if(typeof stream.emit === 'function') {
```

Comment:
Good comment / observation.

It's basically readable vs writable streams. For the purposes of `dust.isStreamable` you need anything that emits events (and hopefully the events contain `data` and `end`). This could even be some custom thing you wrote yourself-- as long as it has an `on` method, Dust can use it as a ReadableStream.

But to `pipe` to a Stream, it needs to be a WritableStream so it needs to have `write` and `end` methods. The `emit` and `on` parts allow piping to conform to Node WritableStreams spec, but they're optional-- for a Stream to be pipable for Dust, it only needs `write` and `end`.

So it isn't sufficient to check `isStreamable` for piping, because you really need a WritableStream, not just any Stream.

If we wanted to restrict the kinds of Streams you can use I'm cool exploring that-- I just tried to make Dust as lax as possible.